AK/Error.h

```diff
@@ -22,35 +23,51 @@ namespace AK {
 
 class Error {
 public:
-    static Error from_errno(int code) { return Error(code); }
-    static Error from_syscall(StringView syscall_name, int rc) { return Error(syscall_name, rc); }
-    static Error from_string_literal(StringView string_literal) { return Error(string_literal); }
+    static Error from_errno(int code, SourceLocation location = SourceLocation::current())
+    {
+        return Error(code, location);
+    }
+
+    static Error from_syscall(StringView syscall_name, int rc, SourceLocation location = SourceLocation::current())
+    {
+        return Error(syscall_name, rc, location);
+    }
+
+    static Error from_string_literal(StringView string_literal, SourceLocation location = SourceLocation::current())
+    {
+        return Error(string_literal, location);
+    }
 
     bool is_errno() const { return m_code != 0; }
     bool is_syscall() const { return m_syscall; }
 
+    SourceLocation location() const { return m_location; }
     int code() const { return m_code; }
     StringView string_literal() const { return m_string_literal; }
 
 protected:
-    Error(int code)
-        : m_code(code)
+    Error(int code, SourceLocation location)
+        : m_location(location)
+        , m_code(code)
     {
     }
 
 private:
-    Error(StringView string_literal)
-        : m_string_literal(string_literal)
+    Error(StringView string_literal, SourceLocation location)
+        : m_location(location)
+        , m_string_literal(string_literal)
     {
     }
 
-    Error(StringView syscall_name, int rc)
-        : m_code(-rc)
+    Error(StringView syscall_name, int rc, SourceLocation location)
+        : m_location(location)
+        , m_code(-rc)
         , m_string_literal(syscall_name)
         , m_syscall(true)
     {
     }
 
+    SourceLocation m_location;
```

Comment:
Are we worried about bloating the size of error type?
I suppose `SourceLocation` is just two pointers and a `u32`.

Reply:
I'm not personally worried, but haven't done the performance analysis to confirm. However, if this becomes an issue there are certainly things we can do to make it smaller. We should be able to get it down to 32 bytes or even less.